d3-chord/index.d.ts

```diff
@@ -1,78 +1,401 @@
+     * target object returned by the source or target accessor, respectively.
+     */
+    endAngle(): (this: This, d: RibbonSubgroupDatum, ...args: any[]) => number;
+    /**
+     * Sets the end angle to a fixed number in radians and returns this ribbon generator.
+     *
+     * @param angle A fixed numeric value for the end angle in radians.
+     */
     endAngle(angle: number): this;
-    endAngle(angle: (this: This, d: ChordSubgroupDatum, ...args: any[]) => number): this;
+    /**
+     * Sets the end angle accessor to the specified function and returns this ribbon generator.
+     *
+     * @param angle An accessor function which is invoked for the source and target of the chord. The accessor function is invoked in the same "this" context as the generator was invoked in and
+     * receives as the first argument the source or target object returned by the respective source or target accessor function of the generator. It is also passed any additional arguments that were passed
+     * into the generator, with the exception of the first element representing the chord datum itself. The function returns the end angle in radians.
+     */
+    endAngle(angle: (this: This, d: RibbonSubgroupDatum, ...args: any[]) => number): this;
+
+    /**
+     * Returns the current rendering context, which defaults to null.
+     */
     context(): CanvasRenderingContext2D | null;
+    /**
+     * Sets the rendering context and returns this ribbon generator.
+     *
+     * If the context is not null, then the generated ribbon is rendered to this context as a sequence of path method calls.
+     *
+     * @param context The rendering context.
+     */
     context(context: CanvasRenderingContext2D): this;
+    /**
+     * Sets the rendering context to null and returns this ribbon generator.
+     *
+     * A path data string representing the generated ribbon will be returned when the generator is invoked with data. See also d3-path.
+     *
+     * @param context null, to remove rendering context.
+     */
     context(context: null): this;
 }
 
-export function ribbon(): RibbonGenerator<any, Chord, ChordSubgroup>;
+/**
+ * Creates a new ribbon generator with the default settings.
+ */
+export function ribbon(): RibbonGenerator<any, Ribbon, RibbonSubgroup>;
+/**
+ * Creates a new ribbon generator with the default settings.
+ *
+ * Accessor functions must be configured for the ribbon generator, should the datum types differ from the defaults.
+ *
+ * The first generic corresponds to the datum type representing a chord for which the ribbon is to be generated. The default type is Chord.
+ *
+ * The second generic corresponds to the datum type of the chord subgroup, i.e. source or target of the cord. The default type is ChordSubgroup.
+ */
 export function ribbon<Datum, SubgroupDatum>(): RibbonGenerator<any, Datum, SubgroupDatum>;
```

Comment:
Is there any chance we want to consider using `RibbonGenerator<{}, Datum, SubgroupDatum>` as the return type of this overload? I believe it would still be compatible with most usages, but it would prevent accidental unsafe usage of a potentially-unknown `this` context.

Again, not a comment for this particular PR. I'm just bringing it up as I notice it.